### ISSUE TITLE
Allow manually adding add-ons to the review queue

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#19986]
We are currently tracking one-off review requests in an external tool, which introduces visibility and prioritization challenges. Those requests should be part of the reviewer tools.

In conjunction with https://github.com/mozilla/addons-server/issues/19985 (and probably following https://github.com/mozilla/addons-server/issues/19984), we'd want a way to add an add-on to the review queue even if there are no versions that have `AWAITING_REVIEW` status.

* On a review page we want to add a way to set and unset a flag for this add-on, indicating that a manual review is needed, along with the due date (from #19985), This is gated by `Reviews:Admin` permission. The default due date should be 3 business days from now.
* When an version is being rejected or approval-confirmed, the flag will automatically be unset.
* The new queue from #19984 should include add-ons that have this flag.

**Comment by diox** [comment]
If we keep the current way of implementing reviewer queues, maybe this could be done with a new `VersionReviewerFlags` `DateTimeField` field called `needs_human_review_before`, which would eventually replace `Version.needs_human_review`.

**Comment by diox** [comment]
Once https://github.com/mozilla/addons-server/issues/19985#issuecomment-1369044064 is implemented, it becomes a lot easier, essentially we'd need to:
- Allow admins to set `due_date` even on approved versions in the current channel they are looking at
- Include approved versions with a non-empty `due_date` in the queue (*)

(*) We still want to exclude versions that are awaiting review and about to be auto-approved though, so we can't just drop the filtering on `status` and replace it with a filter on `due_date` - it likely needs to be an `OR` query, something like `status = 3 OR status = 4 AND due_date IS NOT NULL`

**Comment by eviljeff** [comment]
> This probably needs to be a separate issue that blocks both this issue and #19984.

https://github.com/mozilla/addons-server/issues/20162

**Comment by eviljeff** [comment]
fwiw, I think this is so closely related to #20152 (because both require arbitrary versions to have a `due_date`) it will be addressed at the same time.

**Comment by ioanarusiczki** [comment]
@eviljeff 

- listed addons, auto-approved in the past
https://reviewers.addons.allizom.org/en-US/reviewers/review-listed/617282#review-actions
https://reviewers.addons.allizom.org/en-US/reviewers/review/618517
They showed up in the manual queue , with due dates (default if not set for another date) -> I could reject them, they had no confirm approval like https://reviewers.addons.allizom.org/en-US/reviewers/review-listed/1005848#review-actions
Now they all show up again in the Manual Review queue because I could set again a  new due date. 

- unlisted addons , auto-approved in the past
After setting a due date using the button from More Actions -> show up in the Manual Queue
https://reviewers.addons.allizom.org/en-US/reviewers/review-unlisted/1007344
https://reviewers.addons.allizom.org/en-US/reviewers/review-unlisted/1007498
Once they're in the Manual Review queue they have Confirm Multiple Versions or Block Multiple Versions. Approve Multiple/Reject Multiple are not possible actions. 

- an addon with mixed versions, and already approved 
https://reviewers.addons.allizom.org/en-US/reviewers/review-unlisted/1008143#history
https://reviewers.addons.allizom.org/en-US/reviewers/review/1008143
Shows up in the Manual Queue, I rejected the listed versions. Then I added it back to manual review and it is displayed as 
Impulse Blocker_alsolisted 1.2.5 which redirects to the listed review page https://reviewers.addons.allizom.org/en-US/reviewers/review/1008143. As an admin I can use the URL from Actions to go find the unlisted page for review which as I mentioned above show that block or confirmation of multiple unlisted versions could be done.

**Comment by ioanarusiczki** [comment]
@eviljeff I forgot to mention that an Admin Reviewer needs Reviews:Admin but this permissions without Addons:Review or Addons:ReviewUnlisted cannot access the reviewer tools pages for listed/unlisted addons where the "Set needs human review" is available.

**Comment by eviljeff** [comment]
> @eviljeff I forgot to mention that an Admin Reviewer needs Reviews:Admin but this permissions without Addons:Review or Addons:ReviewUnlisted cannot access the reviewer tools pages for listed/unlisted addons where the "Set needs human review" is available.

That doesn't seem like a new restriction|irritation?

**Comment by eviljeff** [comment]
@ioanarusiczki can you file a follow-up if any actions appear to be missing? I didn't change the logic that controls what actions are available for a version that was neede_human_review

**Comment by eviljeff** [comment]
> * an addon with mixed versions, and already approved
>       https://reviewers.addons.allizom.org/en-US/reviewers/review-unlisted/1008143#history
>       https://reviewers.addons.allizom.org/en-US/reviewers/review/1008143
>       Shows up in the Manual Queue, I rejected the listed versions. Then I added it back to manual review and it is displayed as
>       Impulse Blocker_alsolisted 1.2.5 which redirects to the listed review page https://reviewers.addons.allizom.org/en-US/reviewers/review/1008143. As an admin I can use the URL from Actions to go find the unlisted page for review which as I mentioned above show that block or confirmation of multiple unlisted versions could be done.

Are you saying that the link in the queue is wrong? If so might be something connected to the queue changes @diox was working on?

**Comment by ioanarusiczki** [comment]
@eviljeff I repeated a test with an addon with mixed versions (one version listed, one unlisted) and I it was alright. 

Maybe Andreas is aware about the necessary [permissions ](https://github.com/mozilla/addons-server/issues/19986#issuecomment-1413483923). But if that is not expected, I'm going to file an issue. So please @wagnerand let us know.

**Comment by wagnerand** [comment]
> Maybe Andreas is aware about the necessary https://github.com/mozilla/addons-server/issues/19986#issuecomment-1413483923. But if that is not expected, I'm going to file an issue. So please @wagnerand let us know.

It's fine for now, thanks.

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-130
